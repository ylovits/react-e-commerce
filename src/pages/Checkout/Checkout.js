import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import CheckoutItem from "../../components/CheckoutItem/CheckoutItem";
import StripeBtn from "../../components/StripeBtn/StripeBtn";
import "./Checkout.scss";
import {
  selectCartItems,
  selectCartTotal
} from "../../redux/cart/cartSelectors";
import "./Checkout.scss";

const Checkout = ({ cartItems, cartTotal }) => {
  return (

      <div className="checkout container">
        <div className="checkout-header pb-2 border-bottom justify-content-md-center row">
          <div className="checkout-header-item col-md-3">
            <span>Product</span>
          </div>
          <div className="checkout-header-item col-md-3">
            <span>Quantity</span>
          </div>
          <div className="checkout-header-item col-md-3">
            <span>Price</span>
          </div>
          <div className="checkout-header-item col-md-1">
            <span>Remove</span>
          </div>
        </div>
        <div className="checkout-body pt-2">
          {cartItems.map(cartItem => (
            <CheckoutItem key={cartItem.MPN} cartItem={cartItem} />
          ))}
        </div>
        <div className="checkout-footer pb-2 justify-content-md-center row">
          <span className="text-right col-md-12">TOTAL: {cartTotal}â‚¬</span>
        </div>
        <div className="text-center">
          <h5 className="text-danger">Warning! Test Site. Don't use a real card for payment</h5>
          <p className="text-danger">If you wanna test the payment please use the following test card:</p>
          <p className="text-danger">4242 4242 4242 4242 - Exp: 01-/20 - CW: 123</p>
        </div>
        <div className="text-center">
          <StripeBtn price={cartTotal} data-toggle="modal" data-target="#paymentModal"/>
        </div>
      </div>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  cartTotal: selectCartTotal
});

export default connect(mapStateToProps)(Checkout);
