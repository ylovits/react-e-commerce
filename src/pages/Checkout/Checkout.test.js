import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import Checkout from "./Checkout";

import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});

it("<Checkout /> rendered", () => {
  expect(
    shallow(
      <Provider store={store}>
        <Checkout />
      </Provider>
    )
  ).toMatchSnapshot();
});
