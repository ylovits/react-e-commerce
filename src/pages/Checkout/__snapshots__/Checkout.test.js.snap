// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> rendered 1`] = `ShallowWrapper {}`;
